test/with_dummyserver/test_connectionpool.py

```diff
@@ -166,6 +166,68 @@ def test_connect_timeout(self):
         pool._put_conn(conn)
         self.assertRaises(ConnectTimeoutError, pool.request, 'GET', url, timeout=timeout)
 
+    def test_timeout_method_not_in_whitelist(self):
+        # Requests should time out when the method is not on the retry whitelist
+        block_event = Event()
+        ready_event = self.start_basic_handler(block_send=block_event, num=6)
+
+        # Pool-global timeout
+        timeout = Timeout(read=SHORT_TIMEOUT)
+        pool = HTTPConnectionPool(self.host, self.port, timeout=timeout, retries=True)
+
+        wait_for_socket(ready_event)
+        block_event.clear()
+        self.assertRaises(ReadTimeoutError, pool.request, 'POST', '/')
+        block_event.set() # Release request
+
+        # Request-specific timeouts should raise errors
+        pool = HTTPConnectionPool(self.host, self.port, timeout=LONG_TIMEOUT, retries=True)
+
+        conn = pool._get_conn()
+
+        wait_for_socket(ready_event)
+        now = time.time()
+        self.assertRaises(ReadTimeoutError, pool.request, 'POST', '/', timeout=timeout)
+        delta = time.time() - now
+
+        self.assertTrue(delta < LONG_TIMEOUT, "timeout was pool-level LONG_TIMEOUT rather than request-level SHORT_TIMEOUT")
+        block_event.set() # Release request
+
+        # Timeout int/float passed directly to request and _make_request should
+        # raise a request timeout
+        wait_for_socket(ready_event)
+        self.assertRaises(ReadTimeoutError, pool.request, 'POST', '/', timeout=SHORT_TIMEOUT)
+        block_event.set() # Release request
```

Comment:
@sigmavirus24 Trimming it sounds good.

Removing the patch on connectionpool.py, here's part of the output for `make test`:
`
======================================================================
ERROR: test_timeout_method_not_in_whitelist (test.with_dummyserver.test_connectionpool.TestConnectionPoolTimeouts)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vagrant/urllib3/test/with_dummyserver/test_connectionpool.py", line 180, in test_timeout_method_not_in_whitelist
    self.assertRaises(ReadTimeoutError, pool.request, 'POST', '/')
  File "/opt/smarkets/python/3.5.1/lib/python3.5/unittest/case.py", line 727, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/opt/smarkets/python/3.5.1/lib/python3.5/unittest/case.py", line 176, in handle
    callable_obj(*args, **kwargs)
  File "/vagrant/urllib3/urllib3/request.py", line 70, in request
    **urlopen_kw)
  File "/vagrant/urllib3/urllib3/request.py", line 148, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File "/vagrant/urllib3/urllib3/connectionpool.py", line 682, in urlopen
    **response_kw)
  File "/vagrant/urllib3/urllib3/connectionpool.py", line 653, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/vagrant/urllib3/urllib3/util/retry.py", line 363, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=34364): Max retries exceeded with url: / (Caused by ReadTimeoutError("HTTPConnectionPool(host='localhost', port=34364): Read timed out. (read timeout=0.001)",))
-------------------- >> begin captured logging << --------------------
urllib3.util.retry: DEBUG: Converted retries value: True -> Retry(total=True, connect=None, read=None, redirect=None)
urllib3.connectionpool: DEBUG: Starting new HTTP connection (1): localhost
urllib3.util.retry: DEBUG: Incremented Retry for (url='/'): Retry(total=0, connect=None, read=None, redirect=None)
urllib3.connectionpool: WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None)) after connection broken by 'ReadTimeoutError("HTTPConnectionPool(host='localhost', port=34364): Read timed out. (read timeout=0.001)",)': /
urllib3.connectionpool: DEBUG: Starting new HTTP connection (2): localhost
--------------------- >> end captured logging << ---------------------
`

Would you prefer if I caught the MaxRetryError and explicitly failed the test, instead of letting it fail due to the unexpected raise?